src/silx/test/__init__.py

```diff
@@ -43,6 +44,7 @@ def run_tests(
     :param verbosity: Requested level of verbosity
     :param args: List of extra arguments to pass to `pytest`
     """
+    workdir = tempfile.mkdtemp(prefix="silx-")
```

Comment:
Doc says:

> The user of mkdtemp() is responsible for deleting the temporary directory and its contents when done with it.

IMO best to use `tempfile.TemporaryDirectory` context manager to take care of it